### PR TITLE
Remove spark-test-sugar dependency

### PR DESCRIPTION
**Pull request by MrPowers**
Removing the spark-test-sugar dependency will let us cross compile this library with Scala 2.12.

### CODE REVIEW

File: src/test/scala/examples/fancy_numbers/SparkTest.scala [hunk 1]
```diff
@@ -1,68 +1,78 @@
...
-    val sqlContext = sqlc
-    import sqlContext.implicits._
+    import spark.implicits._
 
     behave like fancyRecordBuilder(records, jc)
 
     "should build records with Spark" in {
       ds.count should be(105)
     }
+
```
Comment (ssimeonov) [inline src/test/scala/examples/fancy_numbers/SparkTest.scala:26]: No need for the extra whitespace for such simple tests: the description and `{ ... }` create sufficient visual separation without wasting vertical space.

**Review by ssimeonov — CHANGES_REQUESTED** [review]

File: src/test/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+log4j.appender.console=org.apache.log4j.ConsoleAppender
+log4j.appender.console.target=System.err
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+log4j.appender.console.layout.ConversionPattern=%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n
+
+# Settings to quiet third party logs that are too verbose
+log4j.logger.org.eclipse.jetty=WARN
+log4j.logger.org.eclipse.jetty.util.component.AbstractLifeCycle=ERROR
+log4j.logger.org.apache.spark.repl.SparkIMain$exprTyper=WARN
+log4j.logger.org.apache.spark.repl.SparkILoop$SparkILoopInterpreter=WARN
```
[the remark this answers is not in the document]
Comment (MrPowers) [reply src/test/resources/log4j.properties]: Fixed, good catch ;)

File: src/test/scala/com/swoop/spark/SparkSessionTestWrapper.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.swoop.spark
+
+import org.apache.spark.sql.SparkSession
+
+trait SparkSessionTestWrapper {
+
+  lazy val spark: SparkSession = {
+    SparkSession
+      .builder()
+      .master("local")
```
[the remark this answers is not in the document]
Comment (MrPowers) [reply src/test/scala/com/swoop/spark/SparkSessionTestWrapper.scala]: Great point, updated to 2 cores & 4 shuffle partitions.

File: src/test/scala/com/swoop/spark/SparkSessionTestWrapper.scala [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      .appName("spark-records")
+      .config(
```
[the remark this answers is not in the document]
Comment (MrPowers) [reply src/test/scala/com/swoop/spark/SparkSessionTestWrapper.scala]: Copied this over from a project using scalafmt.  Updated the code to put this all on one line.  My Scala formatting is the worst, so fine formatting it however you like.

File: src/test/scala/examples/fancy_numbers/SparkTest.scala (hunk 1 above)
Comment (MrPowers) [reply src/test/scala/examples/fancy_numbers/SparkTest.scala]: Yep, agreed, updated!

**Review by ssimeonov — APPROVED** [review]
Great PR, thanks!

**Pull request merged**